File: frontend/src/app/models/domain/school-class.new.dto.ts

export class SchoolClassNewDTO {

    // FIELDS

    private id: number;
    private name: string;
    private start: Date;
    private end: Date;
    private courseId: number;

    // CONSTRUCTOR

    constructor() {}

    // GETTERS / SETTERS

    public getId(): number {
        return this.id;
    }

    public setId(id: number) {
        this.id = id;
    }

    public getName(): string {
        return this.name;
    }

    public setName(name: string) {
        this.name = name;
    }

    public getStart(): Date {
        return this.start;
    }

    public setStart(start: Date) {
        this.start = start;
    }

    public getEnd(): Date {
        return this.end;
    }

    public setEnd(end: Date) {
        this.end = end;
    }

    public getCourseId(): number {
        return this.courseId;
    }

    public setCourseId(courseId: number) {
        this.courseId = courseId;
    }
}

File: backend/src/main/java/com/williamdsw/semsys/domain/dto/SchoolClassNewDTO.java
package com.williamdsw.semsys.domain.dto;

import java.io.Serializable;
import java.util.Date;

import javax.validation.constraints.NotNull;

public class SchoolClassNewDTO implements Serializable
{
	// FIELDS 
	
	private static final long serialVersionUID = 1L;
	private Integer id;
	
	@NotNull(message = "Name is required")
	private String name;
	
	@NotNull(message = "Start date is required")
	private Date start;
	
	@NotNull(message = "End date is required")
	private Date end;
	
	@NotNull(message = "Course is required")
	private Integer courseId;
	
	// CONSTRUCTORS
	
	public SchoolClassNewDTO () {}
	
	// GETTERS / SETTERS
	
	public Integer getId () 
	{
		return id;
	}
	
	public void setId (Integer id) 
	{
		this.id = id;
	}
	
	public String getName () 
	{
		return name;
	}
	
	public void setName (String name) 
	{
		this.name = name;
	}
	
	public Date getStart () 
	{
		return start;
	}
	
	public void setStart (Date start) 
	{
		this.start = start;
	}
	
	public Date getEnd () 
	{
		return end;
	}
	
	public void setEnd (Date end) 
	{
		this.end = end;
	}
	
	public Integer getCourseId () 
	{
		return courseId;
	}
	
	public void setCourseId (Integer courseId) 
	{
		this.courseId = courseId;
	}
}